experimental/packages/opentelemetry-sdk-metrics-base/test/export/ConsoleMetricExporter.test.ts

```diff
@@ -0,0 +1,110 @@
+import { ExportResult } from '@opentelemetry/core';
+import { ConsoleMetricExporter } from '../../src/export/ConsoleMetricExporter';
+import { PeriodicExportingMetricReader } from '../../src/export/PeriodicExportingMetricReader';
+import { ResourceMetrics } from '../../src/export/MetricData';
+import { MeterProvider } from '../../src/MeterProvider';
+import { defaultResource } from '../util';
+import * as assert from 'assert';
+import * as sinon from 'sinon';
+
+
+async function waitForNumberOfExports(exporter: sinon.SinonSpy<[metrics: ResourceMetrics, resultCallback: (result: ExportResult) => void], void>, numberOfExports: number): Promise<ResourceMetrics[]> {
+  if (numberOfExports <= 0) {
+    throw new Error('numberOfExports must be greater than or equal to 0');
+  }
+
+  let totalExports = 0;
+  while (totalExports < numberOfExports) {
+    await new Promise(resolve => setTimeout(resolve, 20));
+    totalExports = exporter.callCount;
+  }
+
+  return [];
+}
+
+/* eslint-disable no-console */
+describe('ConsoleMetricExporter', () => {
+  let previousConsoleDir: any;
+  let exporter: ConsoleMetricExporter;
+  let meterProvider: MeterProvider;
+  let meterReader: PeriodicExportingMetricReader;
+  let meter: metrics.Meter;
+
+  beforeEach(() => {
+    previousConsoleDir = console.dir;
+    console.dir = () => {};
+
+    exporter = new ConsoleMetricExporter();
+    meterProvider = new MeterProvider({ resource: defaultResource });
+    meter = meterProvider.getMeter('ConsoleMetricExporter', '1.0.0');
+    meterReader = new PeriodicExportingMetricReader({
+      exporter: exporter,
+      exportIntervalMillis: 100,
+      exportTimeoutMillis: 100
+    });
+    meterProvider.addMetricReader(meterReader);
+  });
+
+  afterEach(async () => {
+    console.dir = previousConsoleDir;
+
+    await exporter.shutdown();
+    await meterReader.shutdown();
+  });
+
+  it('should export information about span', async () => {
+    const counter = meter.createCounter('counter_total', {
+      description: 'a test description',
+    });
+    const counterAttribute = { key1: 'attributeValue1' };
+    counter.add(10, counterAttribute);
+    counter.add(10, counterAttribute);
+
+    const histogram = meter.createHistogram('histogram', { description: 'a histogram' });
+    histogram.record(10);
+    histogram.record(100);
+    histogram.record(1000);
+
+    const spyConsole = sinon.spy(console, 'dir');
+    const spyExport = sinon.spy(exporter, 'export');
+
+    await waitForNumberOfExports(spyExport, 1);
+    const resourceMetrics = spyExport.args[0];
+    const firstResourceMetric = resourceMetrics[0];
+    const consoleArgs = spyConsole.args[0];
+    const consoleMetric = consoleArgs[0];
+    const keys = Object.keys(consoleMetric).sort().join(',');
+
+    const expectedKeys = [
+      'dataPointType',
+      'dataPoints',
+      'descriptor',
+    ].join(',');
+
+    assert.ok(firstResourceMetric.resource.attributes.resourceKey === 'my-resource', 'resourceKey');
```

Comment:
I can see you add some attributes like "attributeValue1" in the counter and a histogram, it will be good to add checks for those as well.